lib/datadog/tracing/contrib/utils/quantization/http.rb

```diff
@@ -59,22 +59,24 @@ def query(query, options = {})
 
             def query!(query, options = {})
               options ||= {}
-              options[:show] = options[:show] || []
+              options[:obfuscate] = {} if options[:obfuscate] == :internal
+              options[:show] = options[:show] || (options[:obfuscate] ? :all : [])
               options[:exclude] = options[:exclude] || []
 
               # Short circuit if query string is meant to exclude everything
               # or if the query string is meant to include everything
               return '' if options[:exclude] == :all
-              return query if options[:show] == :all
 
-              collect_query(query, uniq: true) do |key, value|
+              query = collect_query(query, uniq: true) do |key, value|
                 if options[:exclude].include?(key)
                   [nil, nil]
                 else
-                  value = options[:show].include?(key) ? value : nil
+                  value = (options[:show] == :all || options[:show].include?(key)) ? value : nil
                   [key, value]
                 end
-              end
+              end unless options[:show] == :all && !(options[:obfuscate] && options[:exclude])
+
+              options[:obfuscate] ? obfuscate_query(query, options[:obfuscate]) : query
```

Comment:
Contrary to what I previously thought, obfuscation does not operate on key value pairs but on the string itself. Therefore it is a separate, second stage, taking place after `collect_query`.